op-gateway/src/OPDisputeGameProofService.ts

```diff
@@ -0,0 +1,150 @@
+import { EVMProofHelper, type IProofService } from '@ensdomains/evm-gateway';
+import { type JsonRpcBlock } from '@ethereumjs/block';
+import { AbiCoder, Contract, JsonRpcProvider, type AddressLike } from 'ethers';
+
+export interface OPProvableBlock {
+  number: number;
+  disputeGameIndex: number;
+}
+
+const DISPUTE_GAME_FACTORY_ABI = [
+  'function gameAtIndex(uint256 _index) external view returns (uint32 gameType_, uint64 timestamp_, address proxy_)',
+  'function gameCount() external view returns (uint256 gameCount_)',
+  // TODO: Use findLatestGames instead
```

Comment:
This will be used to calculate the disputeGameIndex accurately. I couldn't find the DisputeGameFactory contract address in their documentation, but it appears this address is publishing game to the DisputeGameFactory: https://sepolia.etherscan.io/address/0x49277ee36a024120ee218127354c4a3591dc90a9.

Therefore, I will investigate how they post dispute games and proceed accordingly.